java/java.project.ui/src/org/netbeans/spi/java/project/support/ui/CreateJavaClassFileFromClipboard.java

```diff
@@ -0,0 +1,258 @@
+        this.context = context;
+        this.t = t;
+    }
+
+    @Override
+    public Transferable paste() throws IOException {
+        try {
+            Clipboard c = Toolkit.getDefaultToolkit().getSystemClipboard();
+            if (!c.isDataFlavorAvailable(DataFlavor.stringFlavor)) {
+                return t;
+            }
+            String copiedData = (String) c.getData(DataFlavor.stringFlavor);
+            CreateJavaClassFileFromClipboard.ClassContent classContent = extractPackageAndClassName(copiedData);
+            if (classContent == null) {
+                JOptionPane.showMessageDialog(null, "Code not valid to create class"); //NOI18N
+                return t;
+            }
+            Set<FileObject> files = this.context.files();
+            if (files.size() != 1) {
+                return t;
+            }
+            String path = files.iterator().next().getPath();
+            File fileName = new File(path + File.separator + classContent.getClassName() + ".java"); //NOI18N
+            if (fileName.exists()) {
+                JOptionPane.showMessageDialog(null, "Cannot create class already present"); //NOI18N
+                return t;
+            }
+            if (!fileName.createNewFile()) {
+                JOptionPane.showMessageDialog(null, "Cannot create file"); //NOI18N
+                return t;
+            }
+
+            if (classContent.getPackageName() != null) {
+                copiedData = removePackage(copiedData, classContent.getPackageName());
+            }
+            try (BufferedWriter bw = new BufferedWriter(new FileWriter(fileName))) {
+                String packageLocation = getPackageNameFromFile(fileName);
+                if (packageLocation != null && !packageLocation.isEmpty()) {
+                    copiedData = "package " + packageLocation + ";\n" + copiedData;// NOI18N
+                }
+                bw.write(copiedData);
+            }
+
+        } catch (UnsupportedFlavorException ex) {
+            Exceptions.printStackTrace(ex);
+        } catch (IOException ex) {
+            Exceptions.printStackTrace(ex);
+        }
+        return t;
+    }
+    
+    
+     private static class DeadlockTask implements Task<CompilationController> {
+
+        JavaSource.Phase phase;
+        CompilationInfo info;
+
+        public DeadlockTask(JavaSource.Phase phase) {
+            assert phase != null;
+            this.phase = phase;
+        }
+
+        public void run(CompilationController info) {
+            try {
+                info.toPhase(this.phase);
+                this.info = info;
+            } catch (IOException ioe) {
+            }
+        }
+
+    }
+
+    private ClassContent extractPackageAndClassName(String copiedData) {
+        JavaCompiler compiler = ToolProvider.getSystemJavaCompiler();
+        StandardJavaFileManager fileManager = compiler.getStandardFileManager(null, null, null);
```

Comment:
Not sure if the file manager is needed here (passing null to getTask instead of a file manager should be fine). But if yes, please wrap it with try-with-resources, so that it is guaranteed to be closed.

Reply:
Addressed.
You are correct file manager is not needed in this case. So removed file manager passed null in compiler.getTask.